tests/tools/combine-tools.test.js

```diff
@@ -0,0 +1,260 @@
+const fs = require('fs');
+const path = require('path');
+const { combineTools } = require('../../scripts/tools/combine-tools');
+const {
+  expectedDataT1,
+  manualToolsWithMissingData,
+  manualToolsToSort,
+  automatedToolsT5,
+  automatedToolsT4,
+  manualToolsT4,
+  automatedToolsT6,
+  automatedToolsT7,
+  automatedToolsT8,
+  manualToolsT8,
+  automatedToolsT9,
+  manualToolsT9,
+  automatedToolsT12,
+  invalidAutomatedToolsT10,
+  manualToolsWithInvalidURLT11,
+  circularTool
+} = require('../fixtures/combineToolsData');
+
+jest.mock('ajv', () => {
+  return jest.fn().mockImplementation(() => ({
+    compile: jest.fn().mockImplementation(() => (data) => data.title !== 'Invalid Tool'),
+  }));
+});
+
+
+jest.mock('ajv-formats', () => {
+  return jest.fn();
+});
+
+jest.mock('../../scripts/tools/tags-color', () => ({
+  languagesColor: [
+    { name: 'JavaScript', color: 'bg-[#57f281]', borderColor: 'border-[#37f069]' },
+    { name: 'Python', color: 'bg-[#3572A5]', borderColor: 'border-[#3572A5]' }
+  ],
+  technologiesColor: [
+    { name: 'Node.js', color: 'bg-[#61d0f2]', borderColor: 'border-[#40ccf7]' },
+    { name: 'Flask', color: 'bg-[#000000]', borderColor: 'border-[#FFFFFF]' }
+  ]
+}));
+
+jest.mock('../../scripts/tools/categorylist', () => ({
+  categoryList: [
+    { name: 'category1', description: 'Sample Category 1' },
+    { name: 'category2', description: 'Sample Category 2' }
+  ]
+}));
+
+const readJSON = (filePath) => JSON.parse(fs.readFileSync(filePath, 'utf-8'));
+
+describe('combineTools function', () => {
+  const toolsPath = path.join(__dirname, '../', 'fixtures', 'tools', 'tools.json');
+  const tagsPath = path.join(__dirname, '../', 'fixtures', 'tools', 'tags.json');
+  const manualToolsPath = path.join(__dirname, '../', 'fixtures', 'tools', 'manual-tools.json');
+  const automatedToolsPath = path.join(__dirname, '../', 'fixtures', 'tools', 'automated-tools.json');
+
+  let manualTools;
+  let automatedTools;
+  let consoleErrorMock;
+
+  beforeAll(() => {
+    manualTools = readJSON(manualToolsPath);
+    automatedTools = readJSON(automatedToolsPath);
+
+    consoleErrorMock = jest.spyOn(console, 'error').mockImplementation(() => {});
+  });
+
```

Comment:
Why aren't we using `resolveAllMocks()` function here in beforeEach function?

Reply:
tests are failing because of it so i have used **clearAllMocks**